Description: Link statically facebook's third-party libraries
 Given Facebook doesn't guarantee a stable ABI for folly nor its other libraries,
 go the HHVM way and link statically to wangle, folly and fbthrift.
Author: Giuseppe Lavagetto <glavagetto@wikimedia.org>
Forwarded: No
Last-Update: 2016-05-17

index 392f2bf..de18b1d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -119,13 +119,6 @@ AC_CHECK_LIB([ssl], [SSL_library_init], [], [AC_MSG_ERROR([Unable to find ssl])]
 AC_CHECK_LIB([z], [gzread], [], [AC_MSG_ERROR([Unable to find zlib])])
 AC_CHECK_LIB([double-conversion],[ceil],[],[AC_MSG_ERROR(
              [Please install double-conversion library])])
-AC_CHECK_LIB([folly],[getenv],[],[AC_MSG_ERROR(
-             [Please install the folly library])])
-AC_CHECK_HEADER([folly/Likely.h], [], [AC_MSG_ERROR(
-[Couldn't find folly, please download from https://github.com/facebook/folly]
-)], [])
-AC_CHECK_LIB([wangle], [getenv], [], [AC_MSG_ERROR(
-             [Please install the wangle library])])
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_STDBOOL
@@ -162,7 +155,8 @@ AC_CHECK_FUNCS([gettimeofday \
 
 LIBS="$LIBS $BOOST_LDFLAGS $BOOST_CONTEXT_LIB $BOOST_FILESYSTEM_LIB \
       $BOOST_PROGRAM_OPTIONS_LIB $BOOST_SYSTEM_LIB $BOOST_REGEX_LIB \
-      -lpthread -pthread -lthriftprotocol"
+      -lpthread -pthread -l:libfolly.a -l:libthriftprotocol.a \
+      -l:libwangle.a -liberty"
 
 AM_PATH_PYTHON([2.6],, [:])
 AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
